src/components/taskDetails/TaskDependency.tsx

```diff
@@ -0,0 +1,113 @@
+import Link from 'next/link';
+import { ChangeEvent, FC, useState } from 'react';
+import classNames from './task-details.module.scss';
+import { dependency } from '@/app/services/taskDetailsApi';
+import { Textarea } from '.';
+
+type Props = {
+    loading: boolean;
+    fetching: boolean;
+    error: boolean;
+    dependencyData: dependency | undefined;
+    navigateToTask: (taskId: string) => void;
+    isEditing: boolean;
+    updatedDependencies: string[];
+    handleChange: (
+        event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+    ) => void;
+};
+
+const TaskDependency: FC<Props> = ({
+    loading,
+    fetching,
+    error,
+    dependencyData,
+    navigateToTask,
+    isEditing,
+    updatedDependencies,
+    handleChange,
+}) => {
+    const [editedDependencies, setEditedDependencies] =
+        useState<string[]>(updatedDependencies);
+
+    const handleDependenciesChange = (
+        event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+    ) => {
+        const { value } = event.target;
+        const updatedDependencies = value
+            .split(',')
+            .map((taskId) => taskId.trim());
```

Comment:
Let's move this to utilities and test for this small function

Reply:
split the components in 2 files